### PR TITLE
codespell: enable tests

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112949` at d999a09b run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/cba161e2e7ef52ed904bad34038a5e8b">codespell</a></li> (python38Packages.codespell)</li>
    <li><a href="https://gist.github.com/1e5657b1293208c97166edbf1123e184">python37Packages.aspell-python</a></li></li>
    <li><a href="https://gist.github.com/92109cfbcd9032403bdb704ed591d2b9">python37Packages.codespell</a></li></li>
    <li><a href="https://gist.github.com/8df2b2c27d8f66fb8a655591340f45fc">python38Packages.aspell-python</a></li></li>
    <li><a href="https://gist.github.com/af0824f946e1bffd12a57f88f86d4aac">python39Packages.aspell-python</a></li></li>
    <li><a href="https://gist.github.com/58aeedbb27d57a06f48294717d9be032">python39Packages.codespell</a></li></li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112949` at d999a09b run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/2f99b54a14e90c99ee9ff8dc6a0103c3">codespell</a></li> (python38Packages.codespell)</li>
    <li><a href="https://gist.github.com/06ce0d8dea35fbf531a362ca38748c18">python37Packages.aspell-python</a></li></li>
    <li><a href="https://gist.github.com/bc51480d24cc251d02627c539181a375">python37Packages.codespell</a></li></li>
    <li><a href="https://gist.github.com/b224c9d7fff06ca92e81396037b721d1">python38Packages.aspell-python</a></li></li>
    <li><a href="https://gist.github.com/382270a13b77ff08867283223ac23310">python39Packages.aspell-python</a></li></li>
    <li><a href="https://gist.github.com/8097c86d7be6566e23a27d82ab5df8ca">python39Packages.codespell</a></li></li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112949` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>codespell (python38Packages.codespell)</li>
<li>python37Packages.aspell-python</li>
<li>python37Packages.codespell</li>
<li>python38Packages.aspell-python</li>
<li>python39Packages.aspell-python</li>
<li>python39Packages.codespell</li>
</ul>
</details>

File: pkgs/development/python-modules/codespell/default.nix [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
+
   pythonImportsCheck = [ "codespell_lib" ];
 
-  meta = {
+  meta = with lib; {
     description = "Fix common misspellings in source code";
     homepage = "https://github.com/codespell-project/codespell";
-    license = with lib.licenses; [ gpl2 cc-by-sa-30 ];
-    maintainers = with lib.maintainers; [ johnazoidberg ];
-    platforms = lib.platforms.all;
```
Comment (JohnAZoidberg) [inline pkgs/development/python-modules/codespell/default.nix:29]: Why remove this?

File: pkgs/development/python-modules/codespell/default.nix [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-    rm -r codespell_lib setup.cfg
-    # test_command assumes too much about the execution environment
-    pytest --pyargs codespell_lib.tests -k "not test_command"
+  checkInputs = [ aspell-python chardet pytestCheckHook pytest-cov pytest-dependency ];
+
+  preCheck = ''
+    export ASPELL_CONF="dict-dir ${aspellDicts.en}/lib/aspell"
   '';
 
+  # tries to run not rully installed script
```
Comment (JohnAZoidberg) [inline pkgs/development/python-modules/codespell/default.nix:20]: `s/rully/fully/`

File: pkgs/development/python-modules/codespell/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/codespell/default.nix:29]: It is set by buildPythonApplication to the correct value.

Comment (JohnAZoidberg) [reply pkgs/development/python-modules/codespell/default.nix:29]: oh okay (y)

**Pull request merged**